iosxr_store_box.py

```diff
@@ -63,32 +63,51 @@ def main(argv):
-    try:
-        opts, args = getopt.getopt(argv, 'b:m:rvht', ['box=', 'message=', 'release', 'verbose', 'help', 'test'])
-    except getopt.GetoptError:
-        print('Input error')
-        print(master_opts)
-        sys.exit(2)
-    for opt, arg in opts:
-        if opt in ('-h', '--help'):
-            print('==> A tool to copy a box to a maven-like repo')
-            print(master_opts)
-            print('==> E.g. iosxrv-x64-vbox/iosxr_store_box.py -b iosxrv-fullk9-x64.box --release, --message "A new box because..."')
-            print('==> E.g. iosxrv-x64-vbox/iosxr_store_box.py -b iosxrv-fullk9-x64.box -r -v -m "Latest box for release."')
-            sys.exit()
-        if opt in ('-b', '--box'):
-            input_box = arg
-        if opt in ('-m', '--message'):
-            message = arg
-            if not message:
-                message = 'No reason for update specified'
-        elif opt in ('-r', '--release'):
-            artifactory_release = True
-        elif opt in ('-v', '--verbose'):
-            verbose = True
-        elif opt in ('-t', '--test'):
-            test = True
+    # Suck in the input BOX and handle errors
+    parser = argparse.ArgumentParser(
+        formatter_class=RawDescriptionHelpFormatter,
+        description='A tool to upload an image to a maven repo like artifactory ' +
+        'using curl, the image typically being a vagrant virtualbox, but could ' +
+        'be anything.\n' +
+        'User can select snapshot or release, the release images get synced to ' +
+        'devhub.cisco.com - where they are available to customers.\n' +
+        'This tool also sends an email out to an email address or an alias to ' +
+        'inform them of the new image.\n' +
+        'It is designed to be called from other tools, like iosxr_ios2vbox.py.\n\n' +
+        'It will rely on the following environment variables to work:\n ' +
+        'ARTIFACTORY_USERNAME\n ' +
+        'ARTIFACTORY_PASSWORD\n ' +
+        'ARTIFACTORY_LOCATION_SNAPSHOT\n ' +
+        'ARTIFACTORY_LOCATION_RELEASE\n ' +
+        'ARTIFACTORY_SENDER\n ' +
+        'ARTIFACTORY_RECEIVER',
+        epilog="E.g.:\n" +
+        "iosxrv-x64-vbox/iosxr_store_box.py -b iosxrv-fullk9-x64.box --release --verbose --message 'A new box because...'\n" +
+        "iosxrv-x64-vbox/iosxr_store_box.py -b iosxrv-fullk9-x64.box --release, --message 'A new box because...'\n"
+        "iosxrv-x64-vbox/iosxr_store_box.py -b iosxrv-fullk9-x64.box -r -v -m 'Latest box for release.'\n")
+
+    parser.add_argument('BOX_FILE',
+                        help='BOX filename')
+    parser.add_argument('-m', '--message', nargs='?', metavar="'New box reason'",
+                        const='No reason for update specified',
+                        help='Optionally specify a reason for uploading this box.')
+    parser.add_argument('-r', '--release', action='store_true',
+                        help='store in appdevci-release where it will get synced to devhug')
```

Comment:
End users won't necessarily be using appdevci-release or devhug (hug?), so maybe this should be something like `'upload to $ARTIFACTORY_LOCATION_RELEASE rather than $ARTIFACTORY_LOCATION_SNAPSHOT'`?